chain/syncer.go

```diff
@@ -403,14 +404,18 @@ func (syncer *Syncer) HandleNewTipSet(ctx context.Context, ci *types.ChainInfo,
 				}
 			}
 		}
-		if err = syncer.syncOne(ctx, parent, ts); err != nil {
-			// While `syncOne` can indeed fail for reasons other than consensus,
-			// adding to the badTipSets at this point is the simplest, since we
-			// have access to the chain. If syncOne fails for non-consensus reasons,
-			// there is no assumption that the running node's data is valid at all,
-			// so we don't really lose anything with this simplification.
-			syncer.badTipSets.AddChain(chain[i:])
-			return err
+		// If first tipset is widened, only `syncOne` it when length of `chain` is greater than 1.
```

Comment:
nit: it woule be good to add to this comment by describing why do it this way.  How about this:
"If the chain has length greater than 1 then we need to sync each tipset in the chain in order to process the chain fully, including the non-widened first tipset"
"If the the chan has length == 1 we can avoid processing the non-widened tipset as a performance optimization because this tipset cannot be heavier than the widened first tipset."

Reply:
OK, added comments.